### PR TITLE
Fix `search_tap` regex/string matching.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes the error introduced in https://github.com/Homebrew/brew/commit/52ff98853068c03b3bfa777932da1da69e35e583#diff-791b01b40fb03181802be9c2315258b2R153.

Not sure if the test I have added is the best way to go, but I didn't find another command which even get's to this stage in `search_tap`. Couldn't even get it to work locally yet as I am getting a GitHub API error, despite having set `HOMEBREW_GITHUB_API_TOKEN`.

### CODE REVIEW

File: Library/Homebrew/cmd/search.rb [hunk 1]
```diff
@@ -112,15 +112,17 @@ def query_regexp(query)
...
-      Thread.new { search_tap(user, repo, rx) }
+      Thread.new { search_tap(user, repo, regex_or_string) }
     end.inject([]) do |results, t|
       results.concat(t.value)
     end
   end
 
-  def search_tap(user, repo, rx)
+  def search_tap(user, repo, regex_or_string)
+    regex = regex_or_string.is_a?(String) ? /^#{regex_or_string}$/ : regex_or_string
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/search.rb]: May need to use `#{Regexp.escape(regex_or_string)}` here.

File: Library/Homebrew/test/test_integration_cmds.rb [hunk 2]
```diff
@@ -707,6 +707,10 @@ def test_search
     assert_predicate desc_cache, :exist?, "Cached file should exist"
   end
 
+  def test_search_remote_tap
+    assert_match "Warning: Failed to search tap: homebrew/not-a-tap.", cmd("search", "homebrew/not-a-tap/not-a-formula")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/test_integration_cmds.rb]: Probably worth adding a unit rather than integration test for this, if that's 🆒

**Comment by MikeMcQuaid** [comment]
Test failing for this: https://travis-ci.org/Homebrew/brew/builds/162568024

**Comment by reitermarkus** [comment]
Added a unit test instead, which I'm letting fail first before pushing the fix again.

File: Library/Homebrew/test/test_integration_cmds.rb (hunk 2 above)
Comment (reitermarkus) [inline Library/Homebrew/test/test_integration_cmds.rb]: Added a unit test instead.

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus!

**Pull request merged**